### ISSUE TITLE
I2P: Host not found

### ISSUE THREAD

**Issue by Vort** [arvidn/libtorrent#7259]
This problem happens because of incorrect refactoring of `http_connection.cpp` file in 592544f30f587294351bafce3dee8f53ff7cf03e commit.

This is how code should look like:
https://github.com/arvidn/libtorrent/blob/417405fc892903c8f850ca88d02bc045e56d2e19/src/http_connection.cpp#L387-L389
And this is what it was changed to:
https://github.com/arvidn/libtorrent/blob/592544f30f587294351bafce3dee8f53ff7cf03e/src/http_connection.cpp#L385-L388

It was still not possible for me to download file after I reverted this change, but
1. There were no `Host not found` errors anymore.
2. I can see `HTTP/1.1 200 OK` reply from tracker now.

Related: https://github.com/arvidn/libtorrent/issues/2530

_libtorrent version (or branch)_: 690881fec9cc069976cd3d909254a87c22f43ce0 + fixes from #7247 and #7248
_platform/architecture_: Windows x64
_compiler and compiler version_: clang 14.0.0

**Comment by jiigen** [comment]
I can confirm same results on:
__libtorrent version (or branch): https://github.com/arvidn/libtorrent/commit/690881fec9cc069976cd3d909254a87c22f43ce0 + fixes_ from https://github.com/arvidn/libtorrent/issues/7247 and https://github.com/arvidn/libtorrent/issues/7248
platform/architecture:  Linux Mint 20.2 Uma x64
compiler and compiler version: gcc 9.4.0_

**Comment by Vort** [comment]
> after a while it started the download and finished it

This is important result, thank you!
I started program once again and after 30 minutes of waiting I saw some activity.
For some reason test program just exited without any additional output, but looks like I've figured out what is happening.
Most likely, outgoing connections are broken, but incoming connections works somehow.
It explains both long waiting times and eventual success (after some remote peer decides to connect to our address).
